### PR TITLE
T3 Lockup Implementation

### PR DESCRIPTION
**Pull request by tomlinton**
- Implements lockups on the server side with configurable lengths/bonus rates
- Refactored some of the logic to accomodate

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: infra/token-transfer-server/src/lib/lockup.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Adds a lockup
+ * @param userId - user id of the user adding the lockup
+ * @param amount - the amount to be locked
+ * @returns {Promise<Lockup>} Lockup object.
+ */
+async function addLockup(userId, amount, data = {}) {
+  const user = await hasBalance(userId, amount)
+
```
Comment (franckc) [inline infra/token-transfer-server/src/lib/lockup.js]: Maybe add a check to ensure amount > 0

File: infra/token-transfer-server/migrations/20191022000751-create-lockup.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      end: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      bonus_rate: {
+        allowNull: false,
+        type: Sequelize.FLOAT
```
[the remark this answers is not in the document]
Comment (tomlinton) [reply infra/token-transfer-server/migrations/20191022000751-create-lockup.js]: Good catch, thanks! I always forgot to update in both places when I change the data types

File: infra/token-transfer-server/src/lib/balance.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  // Sum locked by lockups
+  const lockedAmount = calculateLocked(user.Lockups)
+  logger.info('Tokens in lockup', lockedAmount.toString())
+
+  // Calculate total available tokens
+  const available = vested
+    .plus(lockupEarnings)
+    .minus(transferWithdrawnAmount)
+    .minus(lockedAmount)
+
```
[the remark this answers is not in the document]
Comment (tomlinton) [reply infra/token-transfer-server/src/lib/balance.js]: Good idea!

File: infra/token-transfer-server/src/lib/lockup.js (hunk 1 above)
Comment (tomlinton) [reply infra/token-transfer-server/src/lib/lockup.js]: 👍

**Pull request merged**